apps/test-timer-qei/main.cpp

```diff
@@ -0,0 +1,66 @@
+#include "HAL_PWM.h"
+#include "HAL_QEI.h"
+#include "mbed.h"
+#include "stm32f4xx_hal.h"
+
+Thread pinA_pwm_thread;
+Thread pinB_pwm_thread;
+
+// arbitrary digital pins. Needs to be different from what is being initialized in the QEI
+DigitalOut pinA(PB_4);
+DigitalOut pinB(PB_10);
+DigitalOut led(LED1);
+Serial pc(SERIAL_TX, SERIAL_RX, 9600);
+
+float period       = 6;
+float pulse_width  = 2;
+float pulse_offset = pulse_width / 2;
+
+void pinA_thread() {
+  while (1) {
+    pinA = 0;
+    wait_ms(pulse_width);

```

Comment:
i feel the sleep for threads functionality is more appropriate since this deals with threads